src/prog_server/models/prog_server.py

```diff
@@ -15,24 +19,37 @@ class ProgServer():
     def __init__(self):
         self.process = None
 
-    def run(self, host='127.0.0.1', port=5000, debug=False) -> None:
+    def run(self, host=DEFAULT_HOST, port=DEFAULT_PORT, debug=False, models={}, **kwargs) -> None:
         """Run the server (blocking)
 
-        Args:
+        Keyword Args:
             host (str, optional): Server host. Defaults to '127.0.0.1'.
-            port (int, optional): Server port. Defaults to 5000.
+            port (int, optional): Server port. Defaults to 8555.
             debug (bool, optional): If the server is started in debug mode
+            models (dict[str, PrognosticsModel]): a dictionary of extra models to consider. The key is the name used to identify it. 
         """
-        self.process = app.run(host = host, port = port, debug=debug)        
+        if not isinstance(models, dict):
```

Comment:
This seems slightly un-Pythonic to me in that we actually only care that it has a `[]` operator, not that it is exactly a `dict`, but I'm not sure how to test the former. Ordinarily I would suggest that the Pythonic way is to just try the operation and see if it works, but in this case I can see the value in surfacing the error early.

Reply:
I see what you mean but I'm not sure how to do it. We can't just check for the [] operator because that would include collections like lists and tuples. I did some googling and don't see a clear alternative